### PR TITLE
add support for multiple env files

### LINKED ISSUE #5: Support for multiple transformation files web.{stage}.config based on .env files
In one of my projects where I'm using this adapter I have several staging environments: local, dev, test, pre-prod, prod. Currently when deploying via Azure Pipelines I apply changes to environment variables in \<appSettings\> through XML variable substitution so I set up corresponding variables with matching names on each stage release. 

Maybe this is correct and should be the way to do stage specific settings but in my other .NET projects we create stage specific transformation files, like web.prod.config, that lives next to the default web.config in the repos making the source of truth closer to the code base instead of separating them in the CI/CD flows. 

There's a few different solutions to this that could be implemented but I think checking for stage specific .env files and creating transformation files based on the variables would be a good way to do it.

Example of I would imagine it could work:

**Assume I have the following .env files:**

> `.env`

```
DATABASE_CONNECTIONSTRING=local-connection
SMTP_HOST=local-smtp-host
```

> `.env.prod`

```
DATABASE_CONNECTIONSTRING=prod-connection
SMTP_HOST=prod-smtp-host
```

**After building the project the following .config files would be created:**

> `web.config` 

```
...
<appSettings>
    ...
    <add key="DATABASE_CONNECTIONSTRING" value="local-connection" />
    <add key="SMTP_HOST" value="local-smtp-host" />
</appSettings>
...
```

> `web.prod.config`
```
...
<appSettings>
    ...
    <add key="DATABASE_CONNECTIONSTRING" value="prod-connection" xdt:Transform="SetAttributes(value)" xdt:Locator="Match(key)" />
    <add key="SMTP_HOST" value="prod-smtp-host" xdt:Transform="SetAttributes(value)" xdt:Locator="Match(key)" />
</appSettings>
...
```

Would a feature like this be a good addition to the project or should stage specific .env variables be handled in a different way?

### PR DESCRIPTION
**Pull request by KraXen72**
the xff, origin etc. env vars can be overwritten by the actual .env files if needed.
resolves #5

### CODE REVIEW

**Comment by KraXen72** [comment]
oh right the feature should be probably documented in the readme as well. something to add later, i guess

**Comment by KraXen72** [comment]
please wait with merging before i adjust it to the xml transformation spec

**Comment by abaga129** [comment]
@KraXen72 sounds good. I'll wait until you are ready to merge.  Adding this feature is a great idea.  It's a use case that comes up for me a lot in ASP.net project. I havent had to use this IIS feature yet in Sveltekit but I'm sure I will at some point.  Let me know if I can help in anyway!

**Comment by KraXen72** [comment]
@abaga129 please take a look

**Comment by abaga129** [comment]
@KraXen72 sorry I wasnt able to review it over the weekend. I'll take a look at it today.

File: index.js [hunk 1]
```diff
@@ -67,52 +86,62 @@ export default function (options) {
...
+				} else {
+					console.info(
+						`Didn't include ${envFn} variables in ${wcFilename} (disabled)`
+					)
+				}
+				// XML attributes cannot contain these characters, will result in IIS Error 500.19
+				for (const key in env) {
+					env[key] = xmlEscape(env[key])
+				}
+
```
Comment (abaga129) [inline index.js]: It looks like this is getting formatted strangely on github.  It looks like it's using hard-tabs.  Running prettier on it should fix it.

**Comment by abaga129** [comment]
After looking through it, I think this is a nice change.  I actually like the idea of creating it off of env files instead of in the config.  If you dont mind finishing that formatting then I'll get it merged! :)

**Comment by KraXen72** [comment]
fixed @abaga129

**Pull request merged**